spec/lib/rmt/downloader_spec.rb

```diff
@@ -103,8 +103,72 @@
         it('has correct content') { expect(File.read(filename)).to eq(content) }
       end
     end
-  end
 
+    describe '#download with If-Modified-Since' do
+      let(:cache_dir) { Dir.mktmpdir }
+      let(:repo_dir) { Dir.mktmpdir }
+      let(:downloader) { described_class.new(repository_url: 'http://example.com', local_path: repo_dir, cache_path: cache_dir) }
+      let(:time) { Time.utc(2018, 1, 1, 10, 10, 0) }
+      let(:if_modified_headers) do
+        {
+          'User-Agent' => "RMT/#{RMT::VERSION}",
+          'If-Modified-Since' => 'Mon, 01 Jan 2018 10:10:00 -0000'
+        }
+      end
+      let(:filename) { 'repomd.xml' }
+      let(:downloaded_file) { downloader.download("/#{filename}", use_cache: true) }
+      let(:cached_content) { 'cached_content' }
+      let(:fresh_content) { 'fresh_content' }
+
+      before do
+        cache_dir
```

Comment:
You can use `let!` instead.
https://relishapp.com/rspec/rspec-core/v/3-4/docs/helper-methods/let-and-let

Reply:
Actually, in this case, you do not need this before block at all. You reference cache_dir and repo_dir in another let above, so they get evaluated before the test runs. Please remove the before block.

Reply:
Yeah, but I guess that's why there's a Rubocop rule against `let!` -- with lazy loading it's hard to say what gets initialized when in the code :D